cmd/kube-controller-manager/app/autoscaling.go

```diff
@@ -92,7 +92,6 @@ func startHPAControllerWithMetricsClient(ctx ControllerContext, metricsClient me
 		replicaCalc,
 		ctx.InformerFactory.Autoscaling().V1().HorizontalPodAutoscalers(),
 		ctx.ComponentConfig.HPAController.HorizontalPodAutoscalerSyncPeriod.Duration,
-		ctx.ComponentConfig.HPAController.HorizontalPodAutoscalerUpscaleForbiddenWindow.Duration,
```

Comment:
In "cmd/kube-controller-manager/app/options/hpacontroller.go" you need to mark this flag as deprecated and state in the help message that it has no effect.

Reply:
Done. Also squashed old commits.